diff -Naur net-tools-1.60/config.h net-tools-1.60_new/config.h
--- net-tools-1.60/config.h	1970-01-01 08:00:00.000000000 +0800
+++ net-tools-1.60_new/config.h	2011-05-10 20:16:42.327810000 +0800
@@ -0,0 +1,74 @@
+/*
+* config.h	Automatically generated configuration includefile
+*
+* NET-TOOLS	A collection of programs that form the base set of the
+*		NET-3 Networking Distribution for the LINUX operating
+*		system.
+*
+*		DO  NOT  EDIT  DIRECTLY
+*
+*/
+
+/* 
+ * 
+ * Internationalization
+ * 
+ * The net-tools package has currently been translated to French,
+ * German and Brazilian Portugese.  Other translations are, of
+ * course, welcome.  Answer `n' here if you have no support for
+ * internationalization on your system.
+ * 
+ */
+#define I18N 0
+
+/* 
+ * 
+ * Protocol Families.
+ * 
+ */
+#define HAVE_AFUNIX 1
+#define HAVE_AFINET 1
+#define HAVE_AFINET6 1
+#define HAVE_AFIPX 1
+#define HAVE_AFATALK 1
+#define HAVE_AFAX25 1
+#define HAVE_AFNETROM 1
+#define HAVE_AFROSE 0
+#define HAVE_AFX25 1
+#define HAVE_AFECONET 0
+#define HAVE_AFDECnet 0
+#define HAVE_AFASH 0
+
+/* 
+ * 
+ * Device Hardware types.
+ * 
+ */
+#define HAVE_HWETHER 1
+#define HAVE_HWARC 1
+#define HAVE_HWSLIP 1
+#define HAVE_HWPPP 1
+#define HAVE_HWTUNNEL 1
+#define HAVE_HWSTRIP 1
+#define HAVE_HWTR 1
+#define HAVE_HWAX25 1
+#define HAVE_HWROSE 0
+#define HAVE_HWNETROM 1
+#define HAVE_HWX25 1
+#define HAVE_HWFR 1
+#define HAVE_HWSIT 1
+#define HAVE_HWFDDI 0
+#define HAVE_HWHIPPI 0
+#define HAVE_HWASH 0
+#define HAVE_HWHDLCLAPB 0
+#define HAVE_HWIRDA 1
+#define HAVE_HWEC 0
+
+/* 
+ * 
+ * Other Features.
+ * 
+ */
+#define HAVE_FW_MASQUERADE 0
+#define HAVE_IP_TOOLS 0
+#define HAVE_MII 0
diff -Naur net-tools-1.60/config.make net-tools-1.60_new/config.make
--- net-tools-1.60/config.make	1970-01-01 08:00:00.000000000 +0800
+++ net-tools-1.60_new/config.make	2011-05-10 20:16:42.329786000 +0800
@@ -0,0 +1,35 @@
+# I18N=0
+HAVE_AFUNIX=1
+HAVE_AFINET=1
+HAVE_AFINET6=1
+HAVE_AFIPX=1
+HAVE_AFATALK=1
+HAVE_AFAX25=1
+HAVE_AFNETROM=1
+# HAVE_AFROSE=0
+HAVE_AFX25=1
+# HAVE_AFECONET=0
+# HAVE_AFDECnet=0
+# HAVE_AFASH=0
+HAVE_HWETHER=1
+HAVE_HWARC=1
+HAVE_HWSLIP=1
+HAVE_HWPPP=1
+HAVE_HWTUNNEL=1
+HAVE_HWSTRIP=1
+HAVE_HWTR=1
+HAVE_HWAX25=1
+# HAVE_HWROSE=0
+HAVE_HWNETROM=1
+HAVE_HWX25=1
+HAVE_HWFR=1
+HAVE_HWSIT=1
+# HAVE_HWFDDI=0
+# HAVE_HWHIPPI=0
+# HAVE_HWASH=0
+# HAVE_HWHDLCLAPB=0
+HAVE_HWIRDA=1
+# HAVE_HWEC=0
+# HAVE_FW_MASQUERADE=0
+# HAVE_IP_TOOLS=0
+# HAVE_MII=0
diff -Naur net-tools-1.60/config.status net-tools-1.60_new/config.status
--- net-tools-1.60/config.status	1970-01-01 08:00:00.000000000 +0800
+++ net-tools-1.60_new/config.status	2011-05-10 20:16:42.330803000 +0800
@@ -0,0 +1,91 @@
+#
+# config.in	Configure.sh source for the net-tools.
+#		This file contains the definition of what the support
+#		library contains.  Most of all, it defines which types
+#		of address families and hardware we know of.
+#
+# NET-TOOLS	A collection of programs that form the base set of the
+#		NET-3 Networking Distribution for the LINUX operating
+#		system.
+#
+# Version:	config.in 1.22	(99-04-19)
+#
+# Author:	Fred N. van Kempen, <waltje@uwalt.nl.mugnet.org>
+#		Copyright 1988-1993 MicroWalt Corporation
+#
+#		Bernd 'eckes' Eckenfels <net-tools@lina.inka.de>
+#		Arnaldo Carvalho de Melo <acme@conectiva.com.br>
+#		  GNU gettext - I18N
+#
+#		This program is free software; you can redistribute it
+#		and/or  modify it under  the terms of  the GNU General
+#		Public  License as  published  by  the  Free  Software
+#		Foundation;  either  version 2 of the License, or  (at
+#		your option) any later version.
+#
+= /*
+=  * config.h	Automatically generated configuration includefile
+=  *
+=  * NET-TOOLS	A collection of programs that form the base set of the
+=  *		NET-3 Networking Distribution for the LINUX operating
+=  *		system.
+=  *
+=  *		DO  NOT  EDIT  DIRECTLY
+=  *
+=  */
+*
+*
+*              Internationalization
+*
+* The net-tools package has currently been translated to French,
+* German and Brazilian Portugese.  Other translations are, of
+* course, welcome.  Answer `n' here if you have no support for
+* internationalization on your system.
+*
+bool 'Does your system support GNU gettext?' I18N n
+*
+*
+*             Protocol Families.
+* 
+bool 'UNIX protocol family' HAVE_AFUNIX y
+bool 'INET (TCP/IP) protocol family' HAVE_AFINET y
+bool 'INET6 (IPv6) protocol family' HAVE_AFINET6 y
+bool 'Novell IPX/SPX protocol family' HAVE_AFIPX y
+bool 'Appletalk DDP protocol family' HAVE_AFATALK y
+bool 'AX25 (packet radio) protocol family' HAVE_AFAX25 y
+bool 'NET/ROM (packet radio) protocol family' HAVE_AFNETROM y
+bool 'Rose (packet radio) protocol family' HAVE_AFROSE n
+bool 'X.25 (CCITT) protocol family' HAVE_AFX25 y
+bool 'Econet protocol family' HAVE_AFECONET n
+bool 'DECnet protocol family' HAVE_AFDECnet n
+bool 'Ash protocol family' HAVE_AFASH n
+*
+*
+*            Device Hardware types.
+*
+bool 'Ethernet (generic) support' HAVE_HWETHER y
+bool 'ARCnet support' HAVE_HWARC y
+bool 'SLIP (serial line) support' HAVE_HWSLIP y
+bool 'PPP (serial line) support' HAVE_HWPPP y
+bool 'IPIP Tunnel support' HAVE_HWTUNNEL y
+bool 'STRIP (Metricom radio) support' HAVE_HWSTRIP y
+bool 'Token ring (generic) support' HAVE_HWTR y
+bool 'AX25 (packet radio) support' HAVE_HWAX25 y
+bool 'Rose (packet radio) support' HAVE_HWROSE n
+bool 'NET/ROM (packet radio) support' HAVE_HWNETROM y
+bool 'X.25 (generic) support' HAVE_HWX25 y
+bool 'DLCI/FRAD (frame relay) support' HAVE_HWFR y
+bool 'SIT (IPv6-in-IPv4) support' HAVE_HWSIT y
+bool 'FDDI (generic) support' HAVE_HWFDDI n
+bool 'HIPPI (generic) support' HAVE_HWHIPPI n
+bool 'Ash hardware support' HAVE_HWASH n
+bool '(Cisco)-HDLC/LAPB support' HAVE_HWHDLCLAPB n
+bool 'IrDA support' HAVE_HWIRDA y
+bool 'Econet hardware support' HAVE_HWEC n
+*
+*
+*           Other Features.
+*
+bool 'IP Masquerading support' HAVE_FW_MASQUERADE n
+bool 'Build iptunnel and ipmaddr' HAVE_IP_TOOLS n
+bool 'Build mii-tool' HAVE_MII n
diff -Naur net-tools-1.60/hostname.c net-tools-1.60_new/hostname.c
--- net-tools-1.60/hostname.c	2001-04-09 01:04:23.000000000 +0800
+++ net-tools-1.60_new/hostname.c	2011-05-10 20:15:44.811438000 +0800
@@ -98,6 +98,7 @@
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
 	default:
+		break;
 	}
 	exit(1);
     };
@@ -117,6 +118,7 @@
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
 	default:
+		break;
 	}
 	exit(1);
     };
@@ -174,6 +176,7 @@
 	printf("%s\n", hp->h_name);
 	break;
     default:
+    	break;
     }
 }
 
diff -Naur net-tools-1.60/lib/inet_sr.c net-tools-1.60_new/lib/inet_sr.c
--- net-tools-1.60/lib/inet_sr.c	2000-02-21 05:46:45.000000000 +0800
+++ net-tools-1.60_new/lib/inet_sr.c	2011-05-10 20:15:44.814431000 +0800
@@ -105,6 +105,7 @@
     case 2:
        isnet = 0; break;
     default:
+    	break;
     }
 
     /* Fill in the other fields. */
diff -Naur net-tools-1.60/lib/x25_sr.c net-tools-1.60_new/lib/x25_sr.c
--- net-tools-1.60/lib/x25_sr.c	2000-05-20 21:38:10.000000000 +0800
+++ net-tools-1.60_new/lib/x25_sr.c	2011-05-10 20:15:44.820412000 +0800
@@ -77,7 +77,7 @@
   rt.sigdigits=sigdigits;
 
   /* x25_route_struct.address isn't type struct sockaddr_x25, Why? */
-  memcpy(&rt.address, &sx25.sx25_addr, sizeof(x25_address));
+  memcpy(&rt.address, &sx25.sx25_addr, sizeof(struct x25_address));
 
   while (*args) {
 	if (!strcmp(*args,"device") || !strcmp(*args,"dev")) {
diff -Naur net-tools-1.60/Makefile net-tools-1.60_new/Makefile
--- net-tools-1.60/Makefile	2001-04-15 22:34:31.000000000 +0800
+++ net-tools-1.60_new/Makefile	2011-05-10 20:15:44.822420000 +0800
@@ -118,9 +118,10 @@
 
 SUBDIRS	= man/ $(NET_LIB_PATH)/
 
-ifeq ($(origin CC), undefined)
-CC	= gcc
-endif
+#CC	= armv6z-mediatek-linux-gnueabi-gcc
+CC = armv6z-mediatek451_001_vfp-linux-gnueabi-gcc
+AR = armv6z-mediatek451_001_vfp-linux-gnueabi-ar
+
 LD	= $(CC)
 
 NLIB	= -l$(NET_LIB_NAME)
diff -Naur net-tools-1.60/version.h net-tools-1.60_new/version.h
--- net-tools-1.60/version.h	1970-01-01 08:00:00.000000000 +0800
+++ net-tools-1.60_new/version.h	2011-05-10 20:15:44.825430000 +0800
@@ -0,0 +1 @@
+#define RELEASE "net-tools 1.60"
